src/applications/hca/containers/IDPage.jsx

```diff
@@ -16,10 +19,19 @@ class IDPage extends React.Component {
     focusElement('.va-nav-breadcrumbs-list');
   }
 
-  render() {
+  componentDidUpdate(prevProps) {
+    // there's no need for logged-in users to see this page
+    if (!prevProps.isLoggedIn && this.props.isLoggedIn) {
+      this.props.router.push('/');
+    }
+  }
+
+  getContent = () => {
+    if (this.props.isProfileLoading) {
+      return <LoadingIndicator />;
```

Comment:
We don't want to show the user the ID form until we know that they are not logged in

Reply:
Might be easier to read if this was inlined in the main `render` and gated the different pieces with ternary logic. Since this is a connected component we can also change the name `isProfileLoading` to something more explicit like `showLoadingIndicator`

Reply:
I inlined it in the main `render`. In this case I like the slightly more descriptive `isProfileLoading`. I prefer to have a sense of _why_ we are showing or hiding the loading indicator if possible.

Reply:
The trade off there is that it requires maintainers to read the render function to understand what this component is doing. The props can document the dynamic parts of the component and the mapStatsToProps can show how the app state translates to the view state.

Reply:
Okay, I'll give that a shot and try to get on board with your way of thinking 😁